pkg/lightning/common/util.go

```diff
@@ -368,3 +368,43 @@ func StringSliceEqual(a, b []string) bool {
 	}
 	return true
 }
+
+// AskForConfirmation uses Scanln to parse user input. A user must type in "yes" or "no" and
+// then press enter. It has fuzzy matching, so "y", "Y", "yes", "YES", and "Yes" all count as
+// confirmations. If the input is not recognized, it will ask again. The function does not return
+// until it gets a valid response from the user. Typically, you should use fmt to print out a question
+// before calling askForConfirmation. E.g. fmt.Println("WARNING: Are you sure? (yes/no)")
+func AskForConfirmation() bool {
+	var response string
+	fmt.Println("please type [y/N] to continue:")
+	_, err := fmt.Scanln(&response)
```

Comment:
have i missed something? i've never seen any design doc saying that supporting pre-check will turn Lightning into an interactive-CLI tool

Reply:
+1

Reply:
I add the interactivity because I found lightning will start to import automatically even we had some warning in pre-check.
so the user had to kill lightning and start it again, and if the user(it was me) didn't fix the waring in one time. he must kill lightning and start again and again.

Reply:
we could make the checks fatal, and having `--disable-checks=aaa,bbb,ccc` (or the opposite `--enable-checks=xxx,yyy,zzz`) to toggle the checks. the error message to mention the corresponding CLI paramters and config.

Reply:
Ok, I made the checks fatal and reuse --check-requirements to disable the checks.
Because it's not necessary to skip some checks.